network/common/src/main/java/org/apache/spark/network/util/JavaUtils.java

```diff
@@ -121,4 +123,68 @@ private static boolean isSymlink(File file) throws IOException {
     }
     return !fileInCanonicalDir.getCanonicalFile().equals(fileInCanonicalDir.getAbsoluteFile());
   }
+
+  private static ImmutableMap<String, TimeUnit> timeSuffixes = 
+    ImmutableMap.<String, TimeUnit>builder()
+      .put("us", TimeUnit.MICROSECONDS)
+      .put("ms", TimeUnit.MILLISECONDS)
+      .put("s", TimeUnit.SECONDS)
+      .put("m", TimeUnit.MINUTES)
+      .put("min", TimeUnit.MINUTES)
+      .put("h", TimeUnit.HOURS)
+      .put("d", TimeUnit.DAYS)
+      .build();
+
+  /**
+   * Convert a passed time string (e.g. 50s, 100ms, or 250us) to a time count for
+   * internal use. If no suffix is provided a direct conversion is attempted.
+   */
+  private static long parseTimeString(String str, TimeUnit unit) {
+    String lower = str.toLowerCase().trim();
+    
+    try {
+      String suffix = "";
+      for (String tail: timeSuffixes.keySet()) {
+        int charIdx = lower.length() - tail.length() - 1;
+        if (lower.endsWith(tail) && (charIdx >= 0 && Character.isDigit(lower.charAt(charIdx)))) {
+          suffix = tail;
+          break;
+        }
+      }
```

Comment:
This whole method is really dense. Should we do something like the following instead:
```
private static long parseTimeString(String str, TimeUnit unit) {
  String lower = str.toLowerCase().trim();
  long parsedTime = -1;
  TimeUnit parsedSuffix = null;

  for (String suffix : timeSuffixes.keySet()) {
    if (lower.endsWith(suffix)) {
      try {
        // e.g. "50ms" = 50, "3 min" = 3, "3a50s" = error
        parsedTime = Long.parseLong(str.substring(0, str.length() - suffix.length()).trim());
        if (parsedTime >= 0) {
          parsedSuffix = timeSuffixes.get(suffix)
          break;
        }
      } catch(NumberFormatException e) {
        // continue
      }
    }
  }

  // No valid time suffix found
  if (parsedSuffix == null) {
    throw new SomeException(...)
  }

  return unit.convert(parsedTime, parsedSuffix)
}
```